src/pretalx/schedule/models/schedule.py

```diff
@@ -69,13 +70,25 @@ def unfreeze(self, user=None):
         from pretalx.schedule.models import TalkSlot
         if not self.version:
             raise Exception('Cannot unfreeze schedule version: not released yet.')
+
+        submission_ids = [talk.submission_id for talk in self.talks.all()]
+        new_talks = [
+            t for t in self.event.wip_schedule.talks.all()
+            if t.submission_id not in submission_ids
+        ]
+
         self.event.wip_schedule.talks.all().delete()
         self.event.wip_schedule.delete()
+
         wip_schedule = Schedule.objects.create(event=self.event)
         talks = []
-        for talk in self.talks.all():
+        for talk in itertools.chain(new_talks, self.talks.all()):
```

Comment:
Alternatively, if you use a queryset instead of a list in L75, you could go with the [union](https://docs.djangoproject.com/en/dev/ref/models/querysets/#union) method of Django querysets, I think?